pkg/network/filter/packet_source_linux.go

```diff
@@ -129,6 +129,25 @@ func (p *AFPacketSource) SetBPF(filter []bpf.RawInstruction) error {
 	return p.TPacket.SetBPF(filter)
 }
 
+// NoPacketType is what getPktType returns when addPktType is not turned on.
+const NoPacketType = 0xff
+
+// getPktType gets whether this packet capture was incoming or outgoing.
+// This requires addPktType = true in the options.
+// incoming: unix.PACKET_HOST
+// outgoing: unix.PACKET_OUTGOING
+func getPktType(stats gopacket.CaptureInfo) uint8 {
+	// if addPktType = true, AncillaryData will contain an AncillaryPktType element;
+	// however, it might not be the first element, so scan through.
+	for _, data := range stats.AncillaryData {
+		pktType, ok := data.(afpacket.AncillaryPktType)
+		if ok {
+			return pktType.Type
+		}
+	}
+	return NoPacketType
```

Comment:
I don't think this is necessary, we can just not set `pktInfo.PktType` if it is not available.